### ISSUE TITLE
Run CPython configure with oil-native

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#1165]
creating new issue based on 

- #306 

----

- On my laptop I got  an error in `parser::setup()` ?  Trying to parse invalid code?

- On desktop, I guess there are different libraries, so we actually can run some stuff, but it's run incorrectly

```
$ ../../_bin/cxx-dbg/osh_eval -c ../../Python-2.7.13/configure
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for --enable-universalsdk... no
checking for --with-universal-archs... 32-bit
checking MACHDEP... linux2
checking EXTRAPLATDIR... 
checking for --without-gcc... no
checking for --with-icc... no
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/home/andy/git/oilshell/oil/_tmp/conf':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

**Comment by andychu** [comment]
And here is `config.log`

```
/* confdefs.h */
#define _GNU_SOURCE 1
#define _NETBSD_SOURCE 1
#define __BSD_VISIBLE 1
#define _BSD_TYPES 1
#define _DARWIN_C_SOURCE 1
#define _XOPEN_SOURCE 600
#define _XOPEN_SOURCE_EXTENDED 1
#define _POSIX_C_SOURCE 200112L

configure: exit 77
```

**Comment by andychu** [comment]
Prerequisite: `bin/osh_eval.py` has to be able to run Python's configure

**Comment by andychu** [comment]
Current status:

```
andy@andy-VirtualBox:~/git/oilshell/oil/_tmp$ ../_bin/cxx-dbg/osh_eval ../Python-2.7.13/configure
osh_eval: /home/andy/git/oilshell/oil/cpp/leaky_core.h:63: void pyos::SignalState::AddUserTrap(int, builtin_trap::_TrapHandler*): Assertion `!"Not Implemented"' failed.
Aborted (core dumped)
```

**Comment by andychu** [comment]
Done!
